src/canvasdrawer.js

```diff
@@ -0,0 +1,1151 @@
+ * @param {Object} options - Options for this Drawer.
+ * @param {OpenSeadragon.Viewer} options.viewer - The Viewer that owns this Drawer.
+ * @param {OpenSeadragon.Viewport} options.viewport - Reference to Viewer viewport.
+ * @param {Element} options.element - Parent element.
+ * @param {Number} [options.debugGridColor] - See debugGridColor in {@link OpenSeadragon.Options} for details.
+ */
+
+class CanvasDrawer extends $.DrawerBase{
+    constructor(){
+        super(...arguments);
+
+        // Options
+
+
+
+        /**
+         * 2d drawing context for {@link OpenSeadragon.Drawer#canvas} if it's a &lt;canvas&gt; element, otherwise null.
+         * @member {Object} context
+         * @memberof OpenSeadragon.Drawer#
+         */
+        this.context = this.canvas.getContext( '2d' );
+
+        /**
+         * Sketch canvas used to temporarily draw tiles which cannot be drawn directly
+         * to the main canvas due to opacity. Lazily initialized.
+         */
+        this.sketchCanvas = null;
+        this.sketchContext = null;
+
+        // We force our container to ltr because our drawing math doesn't work in rtl.
+        // This issue only affects our canvas renderer, but we do it always for consistency.
+        // Note that this means overlays you want to be rtl need to be explicitly set to rtl.
+        this.container.dir = 'ltr';
+
+        // Image smoothing for canvas rendering (only if canvas is used).
+        // Canvas default is "true", so this will only be changed if user specified "false".
+        this._imageSmoothingEnabled = true;
+
+    }
+
+    /**
+     * @returns {Boolean} true if canvas is supported by the browser, otherwise false
+     */
+    static isSupported(){
+        return $.supportsCanvas;
+    }
+
+    getType(){
+        return 'canvas';
+    }
+
+    /**
+     * create the HTML element (e.g. canvas, div) that the image will be drawn into
+     * @returns {Element} the canvas to draw into
+     */
+    createDrawingElement(){
+        let canvas = $.makeNeutralElement("canvas");
+        let viewportSize = this._calculateCanvasSize();
+        canvas.width = viewportSize.x;
+        canvas.height = viewportSize.y;
+        return canvas;
+    }
+
+    /**
+     * Draws the TiledImages
+     */
+    draw(tiledImages) {
+        var _this = this;
+        this._prepareNewFrame(); // prepare to draw a new frame
+        tiledImages.forEach(function(tiledImage){
+            if (tiledImage.opacity !== 0 || tiledImage._preload) {
+                // _this._updateViewportWithTiledImage(tiledImage);
+                _this._drawTiles(tiledImage);
+            }
+        });
+    }
+
+    /**
+     * @returns {Boolean} True - rotation is supported.
+     */
+    canRotate() {
+        return true;
+    }
+
+    /**
+     * Destroy the drawer (unload current loaded tiles)
+     */
+    destroy() {
+        //force unloading of current canvas (1x1 will be gc later, trick not necessarily needed)
```

Comment:
I figure there's no harm in doing that trick, just in case.

Reply:
Ok, will leave it in